decidim-core/app/forms/translatable_presence_validator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# A custom validator to check for presence in I18n-enabled fields. In order to
+# use it do the following:
+#
+#   validates :my_i18n_field, translatable_presence: true
+#
+# This will automatically check for presence for each of the
+# `available_locales` of the form object (or the `available_locales` of the
+# form's organization) for the given field.
+class TranslatablePresenceValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, _value)
+    available_locales_for(record).each do |locale|
+      translated_attr = "#{attribute}_#{locale}"
+      record.errors.add(translated_attr, :blank) unless record.send(translated_attr).present?
+    end
+  end
+
+  private
+
+  def available_locales_for(record)
```

Comment:
I wasn't sure, it just looked similar. If you think this way is better leave it liek this.